File: Python/sentimental-readability/readability.py
def main():
    text = input("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    grade = round(0.0588 * (letters * 100 / words) - 0.296 * ((sentences * 100 / words)) - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade " + str(grade))

def count_letters(text):

    length = len(text)
    letters = 0

    for i in range(length):
        if text[i].isalpha():
            letters += 1

    return letters

def count_words(text):
    length = len(text)
    words = 1

    for i in range(length):
        if text[i] == ' ':
            words += 1
    return words

def count_sentences(text):
    length = len(text)
    sentences = 0
    for i in range(length):
        if text[i] in ['.', '!', '?']:
            sentences += 1
    return sentences

main()

File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    int grade =
        round(0.0588 * (letters * 100 / (float) words) - 0.296 * ((sentences * 100 / (float) words)) - 15.8);
    // Print the grade level
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    int length = strlen(text);
    int letters = 0;
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    // Return the number of letters in text

    return letters;
}

int count_words(string text)
{
    int length = strlen(text);
    int words = 1;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    // Return the number of words in text
    return words;
}

int count_sentences(string text)
{
    int length = strlen(text);
    int sentences = 0;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    // Return the number of sentences in text
    return sentences;
}
